Protomath/PrototypeProvider.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  PrototypeProvider.swift
+//  Prototope
+//
+//  Created by Jason Brennan on 2015-06-22.
+//  Copyright (c) 2015 Khan Academy. All rights reserved.
+//
+
+import Foundation
+
+struct PrototypeProvider {
+	
+	static let prototypeDirectory = "Early-Math-Prototypes"
+	let prototypes: [Prototype]
+	
+	init() {
+		if let resourcePath = NSBundle.mainBundle().resourcePath {
+			let earlyMathPath = resourcePath + "/" + PrototypeProvider.prototypeDirectory
+			let fileManager = NSFileManager.defaultManager()
+			var errorPtr = NSErrorPointer()
+			let files = fileManager.contentsOfDirectoryAtPath(earlyMathPath, error: errorPtr) as! [String]
+			let filtered = files.filter {
+				var isDirectory: ObjCBool = ObjCBool(false)
+				if fileManager.fileExistsAtPath(earlyMathPath + "/" + $0, isDirectory: &isDirectory) {
+					return isDirectory.boolValue
+				}
+				return false
+			}
+			self.prototypes = filtered.map {
+				Prototype(directoryURL: NSURL(fileURLWithPath: earlyMathPath + "/" + $0, isDirectory: true)!, name: $0)!
+			}
+		} else {
+			self.prototypes = []
+		}
+		
+	}
+	
+	
+	func numberOfPrototypes() -> Int {
```

Comment:
Why these three methods instead of having clients use `prototypes`? i.e.

```
provider.prototypes.count // numberOfPrototypes()
provider.prototypes[index].name // prototypeNameAtIndex
provider.prototypes[index].mainFileURL // pathForPrototypeAtIndex